sdk/communication/azure-communication-identity/tests/_shared/testcase.py

```diff
@@ -48,14 +48,21 @@ def process_response(self, response):
             response['body']['string'] = self._replace_keys(response['body']['string'])
 
         return response
-
+    
     def _replace_keys(self, body):
+        def _replace_recursively(dictionary):
```

Comment:
The previous for loop logic did not work, it actually would not perform nested replacements. 
Recursion felt more elegant because it traverses nested json. We also similarly use a recursive function in the responsereplacer.

Reply:
We do this recursively to sanitize fields that are nested in the JSON response.

Example:
{"identity": {"id": "sanitized"}, "accessToken": {"token": "sanitized",
        "expiresOn": "2021-01-30T22:50:20.3668737+00:00"}}

The id and token fields are nested, but need to be sanitized.